Разметка страниц

Roman3341 opened the pull request.

Roman3341 committed:
Разметил главную страницу

Roman3341 committed:
Перезалил изображения

Из-за формата изображения заменил несколько из них.

Komnatnaya commented on index.html:
```diff
@@ -1,12 +1,315 @@
+    <meta name="viewport" content="width=device-width, initial-scale=1">
     <title>HTML Academy: Девайс</title>
   </head>
+
```
Валидатор ругается

Roman3341 replied:
не могу понять, по какой причине он ругается. Прогнал сам. У меня в это месте всё чисто

Komnatnaya replied:
Не в этом месте, на всю страницу.
Уже не ругается)

Komnatnaya commented on index.html:
```diff
@@ -1,12 +1,315 @@
   <body>
 
-    <p>Репозиторий создан для обучения на профессиональном онлайн‑курсе «<a href="https://htmlacademy.ru/intensive/htmlcss">HTML и CSS. Профессиональная вёрстка сайтов</a>».</p>
+    <header class="page-header">
```
![image](https://github.com/htmlacademy-htmlcss/674811-device-1/assets/20788716/c637c08a-33f7-4473-b906-324b79479623)
Модификатор без родительского класса

Roman3341 replied:
Добавил родительский класс в места, где использовался модификатор. Прогнал через БЭМ-Валидатор. Он не ругался.

Komnatnaya commented on index.html:
```diff
@@ -1,12 +1,315 @@
+        </a>
+        <div class="navigation__account">
+          <form action="" class="navigation__account-form">
+            <input type="text" class="navigation__account-form-input">
```
У инпута поиска есть специальный тип.
У инпута должен быть атрибут name

Roman3341 replied:
замечания исправлены.
а могли мы тут применить относительно новый тег search?

Komnatnaya replied:
По логике могли, но не обязательно.

Komnatnaya commented on index.html:
```diff
@@ -1,12 +1,315 @@
+          <img src="#" alt="Логотип сайта">
+        </a>
+        <div class="navigation__account">
+          <form action="" class="navigation__account-form">
```
Адрес для action https://echo.htmlacademy.ru

Roman3341 replied:
устранил

Komnatnaya commented on index.html:
```diff
@@ -1,12 +1,315 @@
+        <div class="navigation__account">
+          <form action="" class="navigation__account-form">
+            <input type="text" class="navigation__account-form-input">
+            <a href="#" class="navigation__account-form-search">Найти</a>
```
Кнопка отправки формы, это button type=submit

Roman3341 replied:
исправил

Komnatnaya commented on index.html:
```diff
@@ -1,12 +1,315 @@
+            <div class="cart-modal">
+              <ul class="cart-modal__product-list">
+                <li class="cart-modal__product-list-item">
+                  <p class="cart-modal__product-name">Любительская селфи-палка</p>
```
Посмотри в стайлгайд на поповер, и оборачивается фиолетовой рамочной при фокус - это ссылка. т.е. картинку и название нужно обернуть ссылкой

Roman3341 replied:
исправил

Komnatnaya commented on index.html:
```diff
@@ -1,12 +1,315 @@
+                <li class="cart-modal__product-list-item">
+                  <p class="cart-modal__product-name">Любительская селфи-палка</p>
+                  <img src="#" alt="Селфи палка">
+                  <button class="cart-modal__close-button"></button>
```
Кнопке добавить type и подпись, чтобы определить что это за кнопка

Komnatnaya replied:
Вообще желательно всех кнопкам

Roman3341 replied:
Исправил

Komnatnaya commented on index.html:
```diff
@@ -1,12 +1,315 @@
+      <section class="slider">
+        <h2 class="visually-hidden">Лучшие модели</h2>
+        <div class="slider__img-wrapper">
+          <img src="#" alt="Текущая модель">
```
Картинка - часть слайда, она должна быть внутри slider__list-item, на каждый слайд своя картинка

Roman3341 replied:
Исправил

Komnatnaya commented on index.html:
```diff
@@ -1,12 +1,315 @@
+            <p class="slider__item-description">Этот обычный, на первый взгляд, квадрокоптер оснащен мощным лазером,
+              замаскированным под стандартную камеру.</p>
+            <div class="slider__about">
+              <p class="text--line-through">Подробнее</p>
```
Ссылка

Roman3341 replied:
Исправил

Komnatnaya commented on index.html:
```diff
@@ -1,12 +1,315 @@
+              <h3 class="features__description-title">Доставка</h3>
+              <p class="features__description-text text">Мы с удовольствием доставим ваш товар прямо к вашему подъезду
+                совершенно бесплатно! Ведь мы неплохо заработаем, поднимая его на ваш этаж.</p>
+              <img src="./img/delivery-man.svg" alt="Иконка доставщика">
```
Иконка - не контентное изображение

Roman3341 replied:
исправил

Komnatnaya commented on index.html:
```diff
@@ -1,12 +1,315 @@
+          </li>
+          <li class="main-catalog__list-item">
+            <a class="main-catalog__list-link">
+              <img src="./img/qvadracopter.svg" width="132" height="69" class="main-catalog__img"
```
Эти иконки тоже.

Roman3341 replied:
исправил

Komnatnaya commented on index.html:
```diff
@@ -1,12 +1,315 @@
+      <article class="delivery-rare">
+        <button class="delivery-rare__button">
+          <span class="delivery-rare__icon-field">
+            <img src="./img/delivery-truck.svg" class="delivery-rare__truck" alt="Иконка грузовика">
```
Не контентное изображение

Roman3341 replied:
исправил

Komnatnaya commented on index.html:
```diff
@@ -1,12 +1,315 @@
+            компаниями:</p>
+          <ul class="about__company-list">
+            <li class="about__comany-list-item">
+              <a href="#" class="about__company-list-link">Меловые линии</a>
```
Это не ссылки, просто текст

Roman3341 replied:
исправил

Komnatnaya commented on index.html:
```diff
@@ -1,12 +1,315 @@
+            распаковками, а также акции, спецпредложения и вот это вот всё.</p>
+        </div>
+        <div class="subscribe__input-wrapper">
+          <form action="post" class="subscribe__form">
```
post - это метод. Его тоже стоит добавить.
А action - это путь отправки.

Roman3341 replied:
исправил

Komnatnaya commented on index.html:
```diff
@@ -1,12 +1,315 @@
+        </div>
+        <div class="subscribe__input-wrapper">
+          <form action="post" class="subscribe__form">
+            <input type="text" class="subscribe__input">
```
У инпута почты есть специальный тип.
У инпута должен быть атрибут name

Komnatnaya replied:
Пропустил разметку кнопки отправки формы

Roman3341 replied:
исправил

Komnatnaya commented on index.html:
```diff
@@ -1,12 +1,315 @@
+          <form action="post" class="subscribe__form">
+            <input type="text" class="subscribe__input">
+          </form>
+          <p class="subscribe__description text"><span class="subscribe__green-check"></span> Это бесплатно</p>
```
Для галочки не нужен дополнительный тег

Roman3341 replied:
исправил

Komnatnaya commented on index.html:
```diff
@@ -1,12 +1,315 @@
+    </main>
+
+    <footer class="page-footer">
+      <img src="#" alt="Лого магазина желтое">
```
Ссылка

Roman3341 replied:
исправил

Komnatnaya commented on index.html:
```diff
@@ -1,12 +1,315 @@
+        <a href="tel:+78128121212" class="page-footer__phone-number">+7 (812) 812-12-12</a>
+      </div>
+      <div class="page-footer__wrapper">
+        <span class="page-footer__yellow-strip"></span>
```
Можно сделать псевдоэлементом.
Вообще пустых тегов быть не должно

Roman3341 replied:
добавил отдельным элементом, чтобы можно было через space-between распределить 3 элемента на равные расстояния. Пустой тег убрал.

Komnatnaya replied:
Псевдоэлементы тоже могут быть в потоке, и на них тоже будет действовать space-between.

Roman3341 replied:
Точно, я что-то привык, что псевдоэлемент вырывают из потока через абсолютное позиционирование

Komnatnaya commented on index.html:
```diff
@@ -1,12 +1,315 @@
+        <ul class="page-footer__socials-list">
+          <li class="page-footer__socials-list-item">
+            <a href="#" class="page-footer__socials-list-link">
+              <img class="page-footer__socials-list-img" src="#" alt="Телеграм">
```
Соц. иконки - не контентные изображения

Roman3341 replied:
исправил

Komnatnaya commented on index.html:
```diff
@@ -1,12 +1,315 @@
+            </a>
+          </li>
+        </ul>
+        <img src="#" alt="Лого академии">
```
Ссылка.

Roman3341 replied:
исправил

Komnatnaya commented on index.html:
```diff
@@ -1,12 +1,315 @@
+      <img src="#" alt="Лого магазина желтое">
+      <div class="page-footer__wrapper">
+        <p class="page-footer__text">Санкт-Петербург, набережная реки Карповки, 5, литера П.</p>
+        <ul class="page-footer__features-list">
```
Блок с соц. ссылками можно сделать отдельным блоком.

Roman3341 replied:
исправил

Komnatnaya replied:
Не поняла, что именно исправил.
Имелось в виду отдельным блоком по БЭМ. К примеру social, тогда все ссылки будут social__link, все li social__item и т.д. Т.е. соц.ссылки могут быть не только в футере, они могут разместиться где угодно, а пока они - элементы футере, вытащить из него не получится.

Roman3341 replied:
исправил

Roman3341 committed:
Исправил ошибки

Исправил ошибки после 1го ревью разметки главной страницы.

Komnatnaya commented on index.html:
```diff
@@ -1,12 +1,307 @@
+        <div class="navigation__account">
+          <form action="https://echo.htmlacademy.ru" role="search" method="get" class="navigation__account-form">
+            <input type="search" name="site-search" class="navigation__account-form-input">
+            <button type="submit" class="navigation__account-form-search" aria-label="Кнопка поиска">Найти</button>
```
Дополнительная подпись нужна только тем элементам, на которых нет видимого текста. Если в кнопке только иконка, то плоховидящий пользователь не может определить что делает эта кнопка, поэтому нужна скрытая подпись, которую прочитает читался.
Если текст в кнопке есть, то ее можно не подписывать.

Roman3341 replied:
Исправил

Roman3341 committed:
Исправил ошибки в разметке #2

Roman3341 committed:
разметил каталог

Komnatnaya commented on catalog.html:
```diff
@@ -0,0 +1,287 @@
+              </a>
+            </li>
+          </ul>
+          <button type="shop-items__button">Загрузить еще</button>
```
Валидатор ругается

Roman3341 replied:
исправил. Глупая ошибка)

Komnatnaya commented on catalog.html:
```diff
@@ -0,0 +1,287 @@
+            <legend class="filters__title">Bluetooth</legend>
+            <ul class="filters__radio-list">
+              <li class="filters__radio-item">
+                <label class="radio__label">
```
![image](https://github.com/htmlacademy-htmlcss/674811-device-1/assets/20788716/dc8da906-3f64-4b54-bc4d-b1ec5dfba59d)
Нет блока catalog и radio.

Roman3341 replied:
исправил

Komnatnaya commented on catalog.html:
```diff
@@ -0,0 +1,287 @@
+                </label>
+              </li>
+              <li class="filters__radio-item">
+                <label class="radio__label">
```
Для чего вообще делать лейбл элементов отдельного блока?
Отличие одного лейбла от другого можно прописать по модификатору

Roman3341 replied:
исправил

Komnatnaya commented on catalog.html:
```diff
@@ -0,0 +1,287 @@
+      <section class="filters">
+        <h2 class="visually-hidden">Фильтры</h2>
+        <p class="catalog__text-bold">Фильтр:</p>
+        <form action="https://echo.htmlacademy.ru/" method="get" class="filters-form">
```
Форма - это блок.
Элементы внутри формы - элементы этого блока.
Потому что элементы формы не используются за пределами формы.

Roman3341 replied:
Исправил

Komnatnaya commented on catalog.html:
```diff
@@ -0,0 +1,287 @@
+          <label class="filters__title">
+            Стоимость
+            <input type="text" name="cost" class="filters__cost-scale">
+            от 0 до 5000
```
Странная разметка.  На этот элемент будет отдельная демка с разделе декора.

Roman3341 replied:
Я приблизительно разметил, чтобы в 8ой раздел сразу не залетать из-за этого блочка.
Решил хоть что-то оставить.

Komnatnaya commented on catalog.html:
```diff
@@ -0,0 +1,287 @@
+        <header class="sort">
+          <h3 class="catalog__text-bold">Сортировка:</h3>
+          <select class="sort__select" aria-label="Сортировка по параметру">
+            <option value="popular" class="sort__value">Популярные</option>
```
Добавь еще пару option

Roman3341 replied:
Добавил

Roman3341 committed:
Правки разметки

внёс правки в каталоге
добавил заголовок в article в индексе
создал файл style.css
